lms/djangoapps/course_home_api/progress/api.py

```diff
@@ -4,8 +4,200 @@
 
 from django.contrib.auth import get_user_model
 from opaque_keys.edx.keys import CourseKey
+from xmodule.graders import ShowCorrectness
+from datetime import datetime, timezone
 
 from lms.djangoapps.courseware.courses import get_course_blocks_completion_summary
+from dataclasses import dataclass, field
+from typing import Optional, List, Dict, Tuple
+
+
+@dataclass
+class _AssignmentBucket:
+    """Holds scores and visibility info for one assignment type."""
+    assignment_type: str
+    expected_total: int
+    last_grade_publish_date: datetime
+    scores: List[float] = field(default_factory=list)
+    visibilities: List[Optional[bool]] = field(default_factory=list)
+    included: List[Optional[bool]] = field(default_factory=list)
```

Comment:
Nit: edx-platform requires python 3.11 or greater, so you can use the [newer notation style](https://peps.python.org/pep-0585/) for these, e.g. `list[float]`, `list[bool | None]`, etc.